### PR TITLE
Migrate LDAP's install.php to a repair step

### PR DESCRIPTION
**Pull request by ChristophWurst**
Didn't test because I never reset my dev instance. So let's see if CI catches anything :)

### CODE REVIEW

**Comment by ChristophWurst** [comment]
>  There was 1 failure:
> 
> 1) Test\InfoXmlTest::testClasses with data set #17 ('user_ldap')
> Asserting install-migration "OCA\User_LDAP\Migration\SetDefaultProvider" exists
> Failed asserting that false is true.
> 
> /home/runner/work/server/server/tests/lib/InfoXmlTest.php:119

:roll_eyes:

**Review by juliusknorr — APPROVED** [review]
Tested and works with settings up a fresh instance and connecting ldap. Config values are properly set

**Review by nickvergessen — APPROVED** [review]
:elephant:

File: apps/user_ldap/lib/Migration/SetDefaultProvider.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	public function getName(): string {
+		return 'Set default LDAP provider';
+	}
+
+	public function run(IOutput $output): void {
+		$state = $this->config->getSystemValue('ldapIgnoreNamingRules', 'doSet');
+		if ($state === 'doSet') {
+			$this->config->setSystemValue('ldapIgnoreNamingRules', false);
+		}
+
```
Comment (blizzz) [inline apps/user_ldap/lib/Migration/SetDefaultProvider.php:56]: this can be dropped actually since not ignoring them is the default behaviour

**Review by blizzz — CHANGES_REQUESTED** [review]

File: apps/user_ldap/lib/Migration/SetDefaultProvider.php (hunk 1 above)
Comment (ChristophWurst) [reply apps/user_ldap/lib/Migration/SetDefaultProvider.php]: :+1: done, and inlined the helper method as it's now only used from one place

**Review by blizzz — APPROVED** [review]
tested, works as advertized :four_leaf_clover:

**Comment by MorrisJobke** [comment]
retriggered CI because there were some issues over the weekend.

**Pull request merged**